### DISCUSSION TITLE
how get word confidence?

### DISCUSSION THREAD

**Discussion by zh794390558** [coqui-ai/STT#1814] [Q&A]
since support `Confidence score for each word`, but how ?

**Comment by reuben** [comment]
You can use the `--json` command line option of the [Python client](https://github.com/coqui-ai/STT/blob/116029aafef156812f96519d20938ed6eeb18efd/native_client/python/client.py#L108-L111) or the [C++ binary](https://github.com/coqui-ai/STT/blob/116029aafef156812f96519d20938ed6eeb18efd/native_client/args.h#L60-L61). Or you can call the API directly, the `*WithMetadata` functions return the full metadata information for each candidate transcript.

**Reply by zh794390558** [reply]
https://github.com/coqui-ai/STT/blob/116029aafef156812f96519d20938ed6eeb18efd/native_client/python/client.py#L38:5
also does not have word confidence. Just word timeline and transcript confidence.

**Reply by reuben** [reply]
Oh, you're completely right, I don't know how I missed that. Per-letter and per-word confidence scores can be exposed but aren't in the API yet. I removed the line from the main README and will open an issue to track this.

**Reply by remyzerems** [reply]
Up on this. This feature would be much appreciated. Should it be added to the issues as an Enhancement request ?

**Reply by juliandarley** [reply]
any news on word confidence metadata? it would be incredibly helpful.

**Reply by reuben** [reply]
PRs always welcome.

**Reply by juliandarley** [reply]
we (i and my son, Raphael) would love to get more involved with helping coqui, because there are many features we would love to see added, including word confidence, which might not be too difficult, and transfer learning (https://colab.research.google.com/github/coqui-ai/STT/blob/main/notebooks/easy_transfer_learning.ipynb). we will start by reading: https://stt.readthedocs.io/en/latest/BUILDING.html) and https://github.com/coqui-ai/STT/blob/main/CONTRIBUTING.rst. it does look like a pretty decent knowledge of C++ is going to be vital - is that correct?
we could also help with polishing documentation, perhaps starting with something small but nevertheless useful like replacing the mentions of Deep Speech?

**Reply by reuben** [reply]
> it does look like a pretty decent knowledge of C++ is going to be vital - is that correct?

Correct.

>  we could also help with polishing documentation, perhaps starting with something small but nevertheless useful like replacing the mentions of Deep Speech?

That sounds like an easy way to get started and get the flow of contributions, yes! Let me know on Gitter/Matrix if you have any questions.

**Reply by remyzerems** [reply]
> 
> 
> PRs always welcome.

PR added : [Issue #2021](https://github.com/coqui-ai/STT/issues/2021)

**Reply by reuben** [reply]
That's not a PR, that's an issue :) But thanks for filing.
